### PR TITLE
Cross platform

### PR DESCRIPTION
**Pull request by Rans4ckeR**
- Adds an additional .NET6.0 'UniversalGL' build without WinForms. Allows client to be run on any device with .NET6.0 installed (new Rampastring.XNAUI package).
- Improves .NET4.8 compatibility on mono
- Cross platform building (build on Linux, run on Windows and vice versa)

Tested (YR):
- .NET4.8 WindowsGL on mono (WSL2 Ubuntu 20.04)
- .NET6.0 UniversalGL on WSL2 Ubuntu 20.04
- .NET6.0 UniversalGL on Fedora 36.1.5
- .NET6.0 UniversalGL on Windows 11
- All other existing builds on Windows 11

Changes:
- Uses the platform safe SafePath methods for handling file system path operations (new Rampastring.Tools package)
- Replaced WinForms Clipboard with [TextCopy](https://github.com/CopyText/TextCopy)
- Replaced WinForms/System.Drawing with [ImageSharp](https://github.com/SixLabors/ImageSharp)
- Replaced WinForms Screen with XNA GraphicsAdapter.DefaultAdapter
- Replaced WindowsIdentity.GetCurrent().Name with Environment.UserName

Limitations for UniversalGL build:
- Can not use WinForms based methods of WindowManager; HasFocus, Minimize, Maximize, Hide, ...
- Win API is only supported on Windows; FlashWindowEx, LoadCursorFromFile
- Registry is only supported on Windows

TODO:
- The Windows Registry is used by TS. Currently this code is guarded by RuntimeInformation.IsOSPlatform() checks. This will need a replacement to store information on the file system instead.

### CODE REVIEW

**Review by Metadorius — APPROVED** [review]
Tested on Arch Linux under KDE, running OpenGL build under Wine works fine. For some reason I can't get it to work under Mono as it crashes my X11 and trying to connect to CnCNet crashes with `Access denied` when pinging (doesn't w/ sudo) but that was there before the PR.

Also while all other stuff is merged or reviewed could you look into handling crash messages without WinForms so we can run the .NET 6 build natively on Linux? I think that's the only blocker.

To recap the conversation from Discord, the suggested way to do this is:
1) for all errors except XNA init or pre-init errors display an XNAUI messagebox, probably with clickable links but that is an extra;
2) to handle the errors that happen pre-init or on init just open the crash log text file using `Process.Start`, can also add the error message on top of the file, trailing with the actual log.

**Comment by Rans4ckeR** [comment]
> To recap the conversation from Discord, the suggested way to do this is:
> 
> 1. for all errors except XNA init or pre-init errors display an XNAUI messagebox, probably with clickable links but that is an extra;
> 2. to handle the errors that happen pre-init or on init just open the crash log text file using `Process.Start`, can also add the error message on top of the file, trailing with the actual log.

This should be implemented now.
